src/common/entity/compute_state_display.ts

```diff
@@ -30,34 +30,19 @@ export const computeStateDisplay = (
   if (domain === "input_datetime") {
     let date: Date;
     if (!stateObj.attributes.has_time) {
-      date = new Date(
-        stateObj.attributes.year,
-        stateObj.attributes.month - 1,
-        stateObj.attributes.day
-      );
+      date = new Date(compareState);
```

Comment:
What is the format of the state?

Reply:
If it is not an ISO datetime string, this will not work in some browsers, like Safari.

Reply:
Formats of `input_datetime`:
Date: YYYY-MM-DD
Time: HH:MM:SS
Datetime: YYYY-MM-DD HH:MM:SS

Reg. Safari: I had taken that into account for datetime, but not for date. I now replaced the Date() constructor / date parsing, so it should be safe now.